homeassistant/components/melcloud/__init__.py

```diff
@@ -66,7 +66,15 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Establish connection with MELClooud."""
     conf = entry.data
-    mel_devices = await mel_devices_setup(hass, conf[CONF_TOKEN])
+    try:
+        mel_devices = await mel_devices_setup(hass, conf[CONF_TOKEN])
+    except ClientResponseError as ex:
+        if isinstance(ex, ClientResponseError) and ex.code == 401:
```

Comment:
We don't need to check the instance type as the `except` statement above catches only this type.